File: backend/src/main/java/com/rungo/api/global/util/CookieUtil.java
package com.rungo.api.global.util;

import jakarta.servlet.http.Cookie;
import jakarta.servlet.http.HttpServletResponse;

public class CookieUtil {

    public static void addCookie(HttpServletResponse response, String name, String value, int maxAge) {

        Cookie cookie = new Cookie(name, value);
        cookie.setPath("/");
        cookie.setHttpOnly(true);
        cookie.setDomain("localhost");
        cookie.setSecure(false); // 로컬에서는 false로 설정
        cookie.setAttribute("SameSite", "Strict");
        cookie.setMaxAge(maxAge);

        response.addCookie(cookie);
    }

    public static void deleteCookie(HttpServletResponse response, String name) {
        Cookie cookie = new Cookie(name, null);
        cookie.setMaxAge(0);
        cookie.setPath("/");
        cookie.setHttpOnly(true);
        response.addCookie(cookie);
    }
}

File: backend/src/main/java/com/rungo/api/global/util/CookieUtil.kt
package com.rungo.api.global.util

import jakarta.servlet.http.Cookie
import jakarta.servlet.http.HttpServletResponse

object CookieUtil {
    private val isProd: Boolean =
        System.getenv("SPRING_PROFILES_ACTIVE")
            ?.split(",")
            ?.map { it.trim() }
            ?.contains("prod") == true

    fun addCookie(response: HttpServletResponse, name: String, value: String, maxAge: Int) {
        Cookie(name, value).apply {
            path = "/"
            isHttpOnly = true
            secure = isProd
            setAttribute("SameSite", if (isProd) "None" else "Strict")

            if (!isProd) {
                domain = "localhost"
            }

            setMaxAge(maxAge)
        }.also { response.addCookie(it) }
    }

    fun deleteCookie(response: HttpServletResponse, name: String) {
        Cookie(name, null).apply {
            path = "/"
            isHttpOnly = true
            secure = isProd
            setAttribute("SameSite", if (isProd) "None" else "Strict")

            if (!isProd) {
                domain = "localhost"
            }

            maxAge = 0
        }.also { response.addCookie(it) }
    }
}